tt_metal/impl/device/multi_device.hpp

```diff
@@ -21,10 +21,11 @@ class DeviceMesh
 {
 public:
     DeviceGrid device_grid;
-    std::vector<std::pair<int, std::unique_ptr<Device>>> managed_devices;
+    std::map<chip_id_t, Device *> managed_devices;
```

Comment:
chip_id_t is an int, right?  use a vector?

Reply:
for a lot of use cases we create device 0 and device 7, just because of how they are connected in a T3000. Can still be a vector in that case, but I'm cleaning this up as part of the DevicePool changes, so I'll leave this alone for now so multichip CI can pass..